src/transaction/concept/api/thing.rs

```diff
@@ -19,26 +19,35 @@
  * under the License.
  */
 
-use async_trait::async_trait;
+#[cfg(not(feature = "sync"))]
 use futures::stream::BoxStream;
 use typeql_lang::pattern::Annotation;
 
+#[cfg(feature = "sync")]
+use crate::common::stream::BoxStream;
 use crate::{
     common::{box_stream, IID},
     concept::{Attribute, AttributeType, Entity, Relation, RoleType, Thing, ThingType},
     Result, Transaction,
 };
 
-#[async_trait]
-pub trait ThingAPI: Clone + Sync + Send {
+#[cfg_attr(not(feature = "sync"), async_trait::async_trait)]
+pub trait ThingAPI: Sync + Send {
```

Comment:
Requiring `Clone` on a trait makes it not object safe, i.e. such a trait can't be used as a trait object (`&dyn ThingAPI`). We rely on that to greatly simplify the bindings, so this requirement needed to be removed.